src/pathfinding_service/model/token_network.py

```diff
@@ -124,62 +130,41 @@ def _calculate_fees(self) -> None:
                     schedule_out=view_out.fee_schedule_sender,
                     receivable_amount=view_in.capacity,
                 )
-                self._is_valid = False
-                break
+                return False
 
             fee = PaymentWithFeeAmount(amount_with_fees - total)
             total += fee  # type: ignore
 
             self.fees.append(FeeAmount(fee))
 
-    @property
-    def edge_attrs(self) -> Iterable[dict]:
-        return (self.G[node1][node2] for node1, node2 in zip(self.nodes[:-1], self.nodes[1:]))
-
-    def to_dict(self) -> dict:
-        return dict(
-            path=[to_checksum_address(node) for node in self.nodes], estimated_fee=sum(self.fees),
-        )
+        return True
 
-    @property
-    def is_valid(self) -> bool:
-        """ Check capacity and settle timeout
+    def _check_validity(self) -> bool:
```

Comment:
Similarly, we can remove the side effect here:
```suggestion
    def _calc_fees_if_valid(self) -> Optional[FeeAmount]:
```